File: testdata/jennies/builders/references/PythonBuilder/builders/some_pkg.py
import typing
from ..cog import builder as cogbuilder
from ..models import some_pkg
from ..models import other_pkg


class Person(cogbuilder.Builder[some_pkg.Person]):
    _internal: some_pkg.Person

    def __init__(self):
        self._internal = some_pkg.Person()

    def build(self) -> some_pkg.Person:
        """
        Builds the object.
        """
        return self._internal    
    
    def name(self, name: other_pkg.Name) -> typing.Self:    
        self._internal.name = name
    
        return self
    

File: testdata/jennies/builders/references/RustBuilder/src/builders/some_pkg.rs
use crate::cog;
use crate::types::other_pkg;
use crate::types::some_pkg;

#[derive(Debug, Clone)]
pub struct PersonBuilder {
    internal: some_pkg::Person,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl PersonBuilder {
    pub fn new() -> Self {
        Self {
            internal: some_pkg::Person::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for PersonBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl PersonBuilder {
    pub fn name(mut self, name: other_pkg::Name) -> Self {
        self.internal.name = name;

        self
    }
}

impl cog::Builder<some_pkg::Person> for PersonBuilder {
    fn build(&self) -> Result<some_pkg::Person, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
